News letter for June (#92)
fortran-lang/fortran-lang.org: _posts/2020-06-01-Fortran-Newsletter-June-2020.md

---
layout: post
title: "Fortran newsletter: June 2020"
category: newsletter

> author: Jeremie Vandenplas, Brad Richardson, and Milan Curcic

Author: Milan, you should probably be the main author.

Reviewer: I think the order is good as is. You originated the newsletter and Brad made a significant addition. I merely tidied things up.

---

Welcome to the June 2020 edition of the monthly Fortran newsletter.
The newsletter comes out on the first calendar day of every month
and details Fortran news from the previous month.

* [fortran-lang.org](#fortran-lang.org)
* [Fortran Discourse](#fortran-discourse)
* [Standard Library](#fortran-standard-library)
* [Package Manager](#package-manager)
* [Events](#events)

## fortran-lang.org

The Fortran website has been up since mid-April, and we've already got great
feedback from the community already.
In the past month we've updated the [Compilers](/compilers) page which is now
comprehensive and includes all major open source and commercial compilers.
The [Learn](/learn) page has also seen significant updates—it's been
reorganized for easier navigation and currently features a quickstart tutorial,
Fortran books, and other online resources.

If you haven't yet, please explore the website and [let us know](https://github.com/fortran-lang/fortran-lang.org/issues)
if you have any suggestions for improvement.
Specifically, we'll be focusing on the [Learn](/learn) page and its mini-books
in the coming months.
Please help us make them better!

> Here are some specific items that we worked on:

Author: ```suggestion
Here are some specific items that we worked (or are still working) on:
```

* [#90](https://github.com/fortran-lang/fortran-lang.org/pull/90)
WIP: Mini-book on building programs
* [#83](https://github.com/fortran-lang/fortran-lang.org/pull/83)
Improving the structure and navigation of the [Learn](/learn) pages
* [#46](https://github.com/fortran-lang/fortran-lang.org/pull/46)
Build website previews from pull requests

## Fortran Discourse

On May 4 we launched the [Fortran Discourse](https://fortran-lang.discourse.group), an online discussion board
for anything and everything Fortran related.
You can use it discuss the Fortran language, ask for help, announce events and/or personal projects, or just lurk
around. 
There are already quite a few interesting discussions going on.
Join us!

## Fortran Standard Library

> 

Author: ```suggestion

Recently we launched a [website](https://stdlib.fortran-lang.org) for the API documentation of the Fortran Standard Library. 
The [website](https://stdlib.fortran-lang.org) is automaticaly generated by [FORD](https://github.com/Fortran-FOSS-Programmers/ford#readme).
[Code of Conduct](https://stdlib.fortran-lang.org/page/contributing/CodeOfConduct.html), [licence](https://stdlib.fortran-lang.org/page/License.html), and [workflow](https://stdlib.fortran-lang.org/page/contributing/Workflow.html) for contributing to the Fortran Standard Library can also be found on the [website](https://stdlib.fortran-lang.org/).

```

Following [this comment](https://github.com/fortran-lang/fortran-lang.org/pull/92/files#r432571024)

Here's what's new in the Fortran Standard Library:

* [#191](https://github.com/fortran-lang/stdlib/pull/191)
WIP: Function for computing Pearson correlations among elements of
an array in the `stdlib_experimental_stats` module
* [#189](https://github.com/fortran-lang/stdlib/pull/189)
WIP: Procedures for sparse matrices operations. Ongoing discussion on the API can be found
[here](https://github.com/fortran-lang/stdlib/wiki/Stdlib-Sparse-matrix-API).
* [#183](https://github.com/fortran-lang/stdlib/pull/183)
Automatic API-doc generation and deployment of this [stdlib website](https://stdlib.fortran-lang.org)
* [#170](https://github.com/fortran-lang/stdlib/pull/170)
Addition of the new functions `diag`, `eye`, and `trace` functions to make working with
matrices easier.
Read the full specifications [here](https://stdlib.fortran-lang.org/page/specs/stdlib_experimental_linalg.html).

## Package Manager

In this past month support for dependencies between packages has been added
to the [Fortran Package Manager (fpm)](https://github.com/fortran-lang/fpm).

> You can specify either a path to another folder on your machine with an fpm package,

Author: ```suggestion
You can specify either a path to another folder on your machine with a fpm package,
```

Reviewer: "an fpm" is okay here because its pronounced "ef-pi-em". See [this](https://blog.apastyle.org/apastyle/2012/04/using-a-or-an-with-acronyms-and-abbreviations.html) article for example.

Author: Thanks for the explanation and link.

---
layout: post
title: "Fortran newsletter: June 2020"
category: newsletter

> author: Milan Curcic

Reviewer: ```suggestion
author: Jeremie Vandenplas, Brad Richardson, and Milan Curcic
```

---

```fortran
print *, 'Hello, World!'

> ```

Reviewer: I thought this was cute for the very first newsletter, but perhaps unnecessary to repeat in each.

Welcome to this monthly Fortran newsletter.
It will come out on the first calendar day of every month,
detailing Fortran news from the previous month.

* [This website](#this-website)
* [Standard Library](#standard-library)
* [Package Manager](#package-manager)
* [Events](#events)

## This website

If you came to this newsletter from elsewhere, welcome to the new Fortran website.
We built this site mid-April and hope for it to be _the_ home of Fortran on the internet,
which traditionally there hasn't been any to date.
Look around and [let us know](https://github.com/fortran-lang/fortran-lang.github.io/issues)
if you have any suggestions for improvement.
Specifically, [Learn](/learn) and [Packages](/packages) are the pages that
we'll be focusing on in the coming months.
Please help us make them better!

* [#90](https://github.com/fortran-lang/fortran-lang.org/pull/90)
WIP: Mini-book on building programs

> 

Reviewer: ```suggestion

### Fortran Discourse

On May 4 we launched the [Fortran Discourse](https://fortran-lang.discourse.group), an online discussion board
for anything and everything Fortran related.
You can use it discuss the Fortran language, ask for help, announce events and/or personal projects, or just lurk
around. 
There are already quite a few interesting discussions going on.
Join us!

```

## Fortran Standard Library

Here's what's new in Fortran Standard Library:

* [#191](https://github.com/fortran-lang/stdlib/pull/191)
WIP: Function for computing Pearson correlations among elements of
an array in the `stdlib_experimental_stats` module.

* [#189](https://github.com/fortran-lang/stdlib/pull/189)
WIP: Procedures for sparse matrices operations. Ongoing discussion on the API can be found
[here](https://github.com/fortran-lang/stdlib/wiki/Stdlib-Sparse-matrix-API)

* [#183](https://github.com/fortran-lang/stdlib/pull/183)

> Automatic API-doc generation and deployment of this [stdlib website](https://stdlib.fortran-lang.org)

Reviewer: I wonder if this item is worth emphasizing more, perhaps in its own section?

Author: Good point. At least it should have its own paragraph ar the start of this section.

* [#170](https://github.com/fortran-lang/stdlib/pull/170)
Addition of the new functions `diag`, `eye`, and `trace` functions to make working with
matrices easier.
Read the full specifications [here](https://stdlib.fortran-lang.org/page/specs/stdlib_experimental_linalg.html)

## Package Manager

In the past months we've seen the first working implementation of the [Fortran Package Manager (fpm)](https://github.com/fortran-lang/fpm).
Specifically:

* fpm supports three commands:
  - `fpm build`--compiles and links your application and/or library.
  - `fpm test`--runs tests if your package has any test programs.
  - `fpm run`--runs the application if your package has an executable program.
* fpm can build an executable program, a library, or a combination of both.
* fpm can fetch and compile other Fortran libraries using fpm that your project depends on
* Currently only gfortran is supported as the compiler backend. fpm will suport other compilers soon.

Read the [fpm packaging guide](https://github.com/fortran-lang/fpm/blob/master/PACKAGING.md)

> to learn how to build your package with fpm.

Reviewer: This is content from the previous newsletter. We shouldn't repeat it here. Let's remove it.

fpm is still in very early development, and we need as much help as we can get.
Here's how you can help today:

* Try to use it. Does it work? No? Let us know!
* Browse the [open issues](https://github.com/fortran-lang/fpm/issues) and see if you can help implement any fixes or features.
* Adapt your Fortran package for fpm.

> * Improve the documentation.

Reviewer: But a call to action like this I think it's okay to repeat because we need help.

The short term goal of fpm is to make development and installation of Fortran packages with dependencies easier.
Its long term goal is to build a rich and decentralized ecosystem of Fortran packages and create a healthy
environment in which new open source Fortran projects are created and published with ease.

* [#82](https://github.com/fortran-lang/fpm/pull/82)
New functionality to enable `git` dependencies. You can now specify a package, the
path to its git repository, and (optionally) a tag, branch, or commit to use. fpm
will then fetch, compile and link this library with your project, including any
dependencies that package has specified. Check out an example [hello world package](https://gitlab.com/everythingfunctional/hello_fpm)

> that utilizes this functionality.

Reviewer: ```suggestion
that uses this functionality.
```

## Events

* [FortranCon 2020](https://tcevents.chem.uzh.ch/event/12) will take place on July 2-4 in Zurich, Switzerland.
Virtual participation is enabled for both attendees and speakers.
Registration is free and due by June 1, 2020.

> * J3/WG5 joint meeting will take place on October 12-16 in Las Vegas, Nevada.

Reviewer: ```suggestion
* J3/WG5 joint meeting, originally slated for October 12-16 in Las Vegas, Nevada, has been [cancelled](https://mailman.j3-fortran.org/pipermail/j3/2020-May/012034.html).
However, the work on proposals for the Fortran Standard doesn't stop.
```
